cmd/upgrade/app/v1alpha2/0.9.0-to-1.0.0/jiva/jiva_upgrade.go

```diff
@@ -0,0 +1,387 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	errors "github.com/openebs/maya/pkg/errors/v1alpha1"
+	deploy "github.com/openebs/maya/pkg/kubernetes/deployment/appsv1/v1alpha1"
+	pv "github.com/openebs/maya/pkg/kubernetes/persistentvolume/v1alpha1"
+	svc "github.com/openebs/maya/pkg/kubernetes/service/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+)
+
+var (
+	replicaPatch = `{
+		"metadata": {
+		   "labels": {
+			  "openebs.io/version": "@upgrade_version@",
+			  "openebs.io/persistent-volume": "@pv_name@",
+			  "openebs.io/replica": "jiva-replica"
+		   }
+		},
+		"spec": {
+			"selector": {
+				"matchLabels":{
+					"openebs.io/persistent-volume": "@pv_name@",
+					"openebs.io/replica": "jiva-replica"
+				}
+			},
+		   "template": {
+			   "metadata": {
+				   "labels": {
+					   "openebs.io/version": "@upgrade_version@",
+					   "openebs.io/persistent-volume": "@pv_name@",
+					   "openebs.io/replica": "jiva-replica"
+				   }
+			   },
+			  "spec": {
+				 "containers": [
+					{
+					   "name": "@r_name@",
+					   "image": "@replica_image@:@upgrade_version@"
+					}
+				 ],
+				 "affinity": {
+					 "podAntiAffinity": {
+						 "requiredDuringSchedulingIgnoredDuringExecution": [
+							 {
+								 "labelSelector": {
+									 "matchLabels": {
+										 "openebs.io/persistent-volume": "@pv_name@",
+										 "openebs.io/replica": "jiva-replica"
+									 }
+								 },
+					 "topologyKey": "kubernetes.io/hostname"
+							 }
+						 ]
+					 }
+				 }
+			  }
+		   }
+		}
+	 }`
+
+	controllerPatch = `{
+		"metadata": {
+		   "labels": {
+			  "openebs.io/version": "@upgrade_version@"
+		   }
+		},
+		"spec": {
+		   "template": {
+			   "metadata": {
+				   "labels":{
+					   "openebs.io/version": "@upgrade_version@"
+				   }
+			   },
+			  "spec": {
+				 "containers": [
+					{
+					   "name": "@c_name@",
+					   "image": "@controller_image@:@upgrade_version@"
+					},
+					{
+					   "name": "maya-volume-exporter",
+					   "image": "@m_exporter_image@:@upgrade_version@"
+					}
+				 ]
+			  }
+		   }
+		}
+	 }`
+
+	servicePatch = `{
+		"metadata": {
+		   "labels": {
+			  "openebs.io/version": "@upgrade_version@"
+		   }
+		}
+	 }`
+
+	kubeConfigPath = "/home/user/.kube/config"
+
+	deployClient  = deploy.NewKubeClient(deploy.WithKubeConfigPath(kubeConfigPath))
+	serviceClient = svc.NewKubeClient(svc.WithKubeConfigPath(kubeConfigPath))
+	pvClient      = pv.NewKubeClient(pv.WithKubeConfigPath(kubeConfigPath))
+)
+
+func getDeploymentDetails(labels, namespace string) (
+	deployName,
+	containerName,
+	version string,
+	err error,
```

Comment:
We should avoid more than 2 return args.

Reply:
Done